File: leetcode-algorithms/367. Valid Perfect Square/solution.cpp
#include <iostream>





bool isPerfectSquare(int num) {
	int i=1;
	while(num>0){
		num -= i;
		i += 2;
	}
	return num == 0;
}

int main()
{
	int num = 14;
	std::cout<<isPerfectSquare(num)<<std::endl;
	return 0;
}

File: leetcode-algorithms/367. Valid Perfect Square/solution.py
def isPerfectSquare(num):
	"""
	:type num: int
	:rtype: bool
	"""

	i = 1
	while num>0:
		num -= i
		i += 2
	return num == 0




if __name__ == '__main__':
	
	num = 16
	print(isPerfectSquare(num))